scripts/release/package_audit.rb

```diff
@@ -0,0 +1,233 @@
+version_tag = ARGV[0] || "latest"
+
+# Manual list - MAKE SURE UP TO DATE especially base images
+containers = [
+  { name: "ballaerospace/cosmosc2-ruby:#{version_tag}", base_image: "alpine:3.15.0", apk: true, gems: true },
+  { name: "ballaerospace/cosmosc2-node:#{version_tag}", base_image: "ballaerospace/cosmosc2-ruby:#{version_tag}", apk: true },
+  { name: "ballaerospace/cosmosc2-base:#{version_tag}", base_image: "ballaerospace/cosmosc2-ruby:#{version_tag}", apk: true, gems: true },
+  { name: "ballaerospace/cosmosc2-cmd-tlm-api:#{version_tag}", base_image: "ballaerospace/cosmosc2-base:#{version_tag}", apk: true, gems: true },
+  { name: "ballaerospace/cosmosc2-init:#{version_tag}", base_image: "ballaerospace/cosmosc2-base:#{version_tag}", apk: true, gems: true, yarn: "/cosmos/plugins/yarn.lock" },
+  { name: "ballaerospace/cosmosc2-operator:#{version_tag}", base_image: "ballaerospace/cosmosc2-base:#{version_tag}", apk: true, gems: true },
+  { name: "ballaerospace/cosmosc2-script-runner-api:#{version_tag}", base_image: "ballaerospace/cosmosc2-base:#{version_tag}", apk: true, gems: true },
+  { name: "ballaerospace/cosmosc2-redis:#{version_tag}", base_image: "redis:6.2", apt: true },
+  { name: "ballaerospace/cosmosc2-traefik:#{version_tag}", base_image: "traefik:2.4.13", apk: true },
+  { name: "ballaerospace/cosmosc2-minio-init:#{version_tag}", base_image: "minio/mc:RELEASE.2021-12-10T00-14-28Z", rpm: true },
+  { name: "minio/minio:RELEASE.2021-06-17T00-10-46Z", rpm: true },
+]
+
+$overall_apk = []
+$overall_apt = []
+$overall_rpm = []
+$overall_gems = []
+$overall_yarn = []
+
+def make_sorted_hash(name_versions)
+  result = {}
+  name_versions.sort!
+  name_versions.each do |name, version, package|
+    result[name] ||= [[], []]
+    result[name][0] << version
+    result[name][1] << package
+  end
+  result.each do |name, data|
+    data[0].uniq!
+    data[1].uniq!
+  end
+  result
+end
+
+def breakup_versioned_package(line, name_versions, package)
+  split_line = line.split('-')
+  found = false
+  (split_line.length - 1).times do |index|
+    i = index + 1
+    if (split_line[i][0] =~ /\d/) or split_line[i -1] == 'pubkey'
+      name = split_line[0..(i - 1)].join('-')
+      version = split_line[i..-1].join('-')
+      name_versions << [name, version, package]
+      found = true
+      break
+    end
+  end
+  raise "Couldn't breakup version for #{package}" unless found
+end
+
+def extract_apk(container)
+  container_name = container[:name]
+  name_versions = []
+  lines = `docker run --rm #{container_name} apk list -I`
+  lines.each_line do |line|
+    package = line.split(' ')[0]
+    breakup_versioned_package(package, name_versions, package)
+  end
+  $overall_apk.concat(name_versions)
+  make_sorted_hash(name_versions)
+end
+
+def extract_apt(container)
+  container_name = container[:name]
+  results = `docker run --rm #{container_name} apt list --installed`
```

Comment:
Didn't really think about using apt since we're on Alpine. How does this come into play?

Reply:
The redis container is debian based.